joint_limits/test/test_simple_joint_limiter.cpp

```diff
@@ -12,31 +12,266 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/// \author Dr. Denis Stogl
+/// \author Dr. Denis Stogl, Guillaume Walck
 
-#include <gmock/gmock.h>
+#include "test_simple_joint_limiter.hpp"
 
-#include <memory>
-#include <string>
+TEST_F(SimpleJointLimiterTest, when_loading_limiter_plugin_expect_loaded)
+{
+  // Test SimpleJointLimiter loading
+  ASSERT_NO_THROW(
+    joint_limiter_ = std::unique_ptr<JointLimiter>(
+      joint_limiter_loader_.createUnmanagedInstance(joint_limiter_type_)));
+  ASSERT_NE(joint_limiter_, nullptr);
+}
+
+/* FIXME: currently init does not check if limit parameters exist for the requested joint
+TEST_F(SimpleJointLimiterTest, when_joint_not_found_expect_init_fail)
+{
+  SetupNode("simple_joint_limiter");
+  Load();
 
-#include "joint_limits/joint_limiter_interface.hpp"
-#include "joint_limits/joint_limits.hpp"
-#include "pluginlib/class_loader.hpp"
-#include "rclcpp/executor.hpp"
+  if (joint_limiter_)
+  {
+    // initialize the limiter
+    std::vector<std::string> joint_names = {"bar_joint"};
+    ASSERT_FALSE(joint_limiter_->init(joint_names, node_));
+  }
+}
+*/
 
-TEST(TestLoadSimpleJointLimiter, load_limiter)
+/*FIXME: currently dt is not tested and leads to nan which lets all other tests pass
```

Comment:
What is the issue exactly here? Why do test pass if solution is nan?

Reply:
because acceleration being nan, does not affect the limitation in that particular testcase where dt was null (by mistake). The problem was subsequent integration with dt being null, which did not give the expected result. This is how I found the issue.

More generally I think it is easier to add a new if statement to fail the limitation if dt is null.